### DISCUSSION TITLE
Field Default value not inserted when adding new record in Table grid

### DISCUSSION THREAD

**Discussion by gmuskie** [pawelsalawa/sqlitestudio#5095] [Q&A]
### Details
A Default value for a filed has been specified in the stucture. When adding new record in Tables grid, nothing happend. It showed NULL value. However, no problem when running in SQL Editor.

### Steps to reproduce

![image](https://github.com/user-attachments/assets/6ed2069d-d43a-4515-a947-6414dbb6a5a2)

![image](https://github.com/user-attachments/assets/330cab0c-6fbf-46df-b018-e9307530cb7c)

![image](https://github.com/user-attachments/assets/18da6066-7e96-4e5e-857f-b783e7451112)

![image](https://github.com/user-attachments/assets/c87ba85f-22da-45cc-b4e6-6e63a477d437)

![image](https://github.com/user-attachments/assets/0eddabeb-3a29-48df-8cd7-9dcc67388651)

### Operating system
Windows 10 Pro 22H2

### SQLiteStudio version
v3.4.4

**Comment by pawelsalawa** [comment]
The `DEFAULT` is used by SQLite when value for certain column is not provided during insertion.

When you insert from table view, the whole row of data is inserted. All columns. It is you, who enter values for the row before commit. NULL value is correct for your column and SQLiteStudio must allow you to insert NULL.

If you want to enforce default values for a column during insertion of NULL values, add `NOT NULL` constraint to your `DEFAULT` column. It will enforce SQLiteStudio to use DEFAULT when value is NULL.

**Comment by marcpley** [comment]
I was about to ask the same question. Thanks gmuskie.
Pawel, I understand your answer, which is logical from a SQLite database point of view.
But for user, the default values don't appear in the grid until committing the insertion AND refresh the data.
This could be solved with an option like that : 
![INSERT](https://github.com/user-attachments/assets/799760c7-3ce6-4e90-8386-6b7c04fed746)
